### PR TITLE
debounce search, move to twins view

### PR DESCRIPTION
**Pull request by karladler**

### CODE REVIEW

File: dapps/assets/src/components/DataContainer.ts [hunk 1]
```diff
@@ -48,14 +48,22 @@ export default class DataContainerComponent extends mixins(EvanComponent) {
...
+  mounted() {
+    this.searchTerm = this.$route.params.query || '';
+  }
+
+  async initialQuery(searchTerm = null) {
     this.isLoading = true;
     this.page = 0;
     this.data = [];
-    const { result, total } = await this.search.query(this.type, { searchTerm: this.searchTerm });
+    this.searchTerm = searchTerm;
```
Comment (datrinh) [inline dapps/assets/src/components/DataContainer.ts:51]: Is this side effect necessary?
Before ```this.searchTerm = to.params.query;``` is called

File: dapps/assets/src/components/digitaltwins/DigitalTwins.scss [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    cursor: pointer;
+  }
+
+  & input {
+    padding: 2px 8px;
+    outline: none;
+  }
+}
+
+/deep/ .evan-table {
```
Comment (datrinh) [inline dapps/assets/src/components/digitaltwins/DigitalTwins.scss]: evan-table component is available now. These styles can be removed.

File: dapps/assets/src/components/digitaltwins/DigitalTwins.ts [hunk 3]
```diff
@@ -48,6 +50,15 @@ export default class DigitalTwinsComponent extends mixins(EvanComponent) {
     required: true
   }) fetchMore: Function;
 
+  @Prop({
+    required: true
+  }) search: Function;
+
+  @Watch('searchTerm')
+    onSearchtermChanged(searchTerm: string, oldVal: string) {
```
Comment (datrinh) [inline dapps/assets/src/components/digitaltwins/DigitalTwins.ts]:
```suggestion
    onSearchTermChanged(searchTerm: string, oldVal: string) {
```
Fix function name accordingly

File: dapps/assets/src/components/digitaltwins/DigitalTwins.ts [hunk 4]
```diff
@@ -66,5 +77,17 @@ export default class DigitalTwinsComponent extends mixins(EvanComponent) {
...
\ No newline at end of file
+  );
+
+  async searchHandler () {
+    if (typeof this.search === 'function') {
+      this.search(this.searchTerm);
+    }
+  }
+
+  searchHandlerDebounced = debounce(
```
Comment (datrinh) [inline dapps/assets/src/components/digitaltwins/DigitalTwins.ts:113]: Why not debounce in searchHandler and switch depending on debounce option parameter?

**Review by datrinh — CHANGES_REQUESTED** [review]

File: dapps/assets/src/components/DataContainer.ts (hunk 1 above)
Comment (karladler) [reply dapps/assets/src/components/DataContainer.ts:51]: yes, bc we call this always when the searchTerm changed, not only once after mounting

**Review by datrinh — APPROVED** [review]

**Pull request merged**